### ISSUE TITLE
Issue with scrolling on drag start and clear selected area

### ISSUE THREAD

**Issue by shaneharter** [DominicTobias/react-image-crop#173]
Hello, 

Thank you for your work here. I could not find a Donate button but will gladly support the project if I use it in production. 

My issue comes when testing ReactCrop on a large image (1295 x 3423). When I start a drag, or clear the selected area with a click the screen is automatically scrolled to the top of the image. This only happens once on each of the events. I looked through the code for anything explicit to do with scrolling image into view and didn't see anything. I'm happy to continue debugging but was curious if you had seen this before? 

I'm on macos chrome 65, running package version 3.0.11

Thanks.

**Comment by DominicTobias** [comment]
Hi, thanks for the kind words.

I actually can't replicate your issue on Chrome 65 macos. Here's what I did:

- `npm start` in this repo.
- open demo/index.html
- Used a 1900x3000 (tall) image
- Drew a crop and scrolled to bottom as I was drawing it to make it a lot longer than screen viewport
- Observed no scroll changes
- Click the two "Programatically set..." buttons at the button too and observed that they also didn't change the scroll.

(Additionally I commented out the `crop: { .. }` part of the setState in onImageLoaded as well as the `maxHeight: 80` just to give me some more drawing flexibility.)

Maybe take a look at what's being re-rendered when you change crop (or what else is happening in your app). It's normal for ReactCrop element to re-render but maybe there's something else making it jump.

**Comment by shaneharter** [comment]
I can re-create in your example if I move the crop-editor down the page a bit. I would fork to show you but it's a 1 LOC change: 

-  <div id="crop-editor"></div>
+  <div style="height: 100px">Space Goes Here</div>
+  <div id="crop-editor"></div>

**Comment by DominicTobias** [comment]
Here's what I'm doing:

https://im5.ezgif.com/tmp/ezgif-5-fa9d632dac.gif

**Comment by shaneharter** [comment]
![](https://media.giphy.com/media/fMzxX4uHiVpz0F5vsf/giphy.gif)

**Comment by andychisholm** [comment]
+1 I'm seeing the same problem. Similar issue raised back in Aug, 17:
https://github.com/DominicTobias/react-image-crop/issues/111

Scroll only occurs if the component is below the current scroll height. I.e. if you scroll past the top of the component and then begin cropping, no scrolling will happen. Otherwise, if you're scrolled above the top of the component, the component will snap to the top of the scroll window view on mousedown.

**Comment by andychisholm** [comment]
The culprit appears to be auto-focus on the component, i.e.:
https://github.com/DominicTobias/react-image-crop/blob/master/dist/ReactCrop.js#L716

**Comment by shaneharter** [comment]
Correct, I've fixed it temporarily by adding this... have not found time to PR yet (this project is still under development)

```
   -   _this.componentRef.focus();
   +    _this.componentRef.focus({
   +     preventScroll: true
   +   });
```

**Comment by DominicTobias** [comment]
Great find Shane! I’ve had trouble replicating this in the past but Andy’s
instructions sound clear and will give it a go and checkout this property
you found 👌

**Comment by DominicTobias** [comment]
Released in v4.0.0 let me know if you have any problems, as mentioned this is only a breaking change for people who had overridden `$mobile-media-query` scss variable

**Comment by andychisholm** [comment]
Awesome, thanks @DominicTobias!
